server/src/app.ts

```diff
@@ -1,204 +1,253 @@
-    return user;
-  }
+        return user;
+    }
 
-  let user: any = await getUserById(User, id.id);
+    let user: any = await getUserById(User, id.id);
 
     if (!user) {
-      res.send({ message: "this user does not exist" });
+        res.send({ message: "this user does not exist" });
     }
-  
+
     res.send({ user: user });
-    
 });
 
 const getPersonDetails = app.get("/api/person-details/:id", async (req, res, next) => {
-  const id = req.params;
+    const id = req.params;
 
     if (!id) {
-      res.send({ message: "something went wrong big time" });
+        res.send({ message: "something went wrong big time" });
+    } else {
+        console.log(id.id)
     }
 
-  async function getPersonById(model: typeof Person, id: any) {
-    let person: any = await 
-    model.findById(id)
-              .populate('memories')
-    return person;
-  }
+    async function getPersonById(model: typeof Person, id: any) {
+        let person: any = await 
 
-  let person: any = await getPersonById(Person, id.id);
+        model.findById(id)
+                    .populate('memories')
 
-      if (!person) {
-        res.send({ message: "this user does not exist" });
-      }
+        return person;
+    }
 
-  res.send(person);
+    let person: any = await getPersonById(Person, id.id);
+
+    if (!person) {
+        res.send({ message: "this user does not exist" });
+    } else {
+        res.send(person);
+    }
 });
 
 const addMemory = app.post("/api/add-memory/:id", async (req, res, next) => {
-  const id = req.params;
-  const userInput = req.body;
-
-  if (!id) {
-    return res.send({ message: "something went wrong big time" });
-  }
-
-  if (!userInput.title) {
-    return res.send({ message: "something went wrong big time - title" });
-  }
-
-  if (!userInput.text) {
-    return res.send({ message: "something went wrong big time - text" });
-  }
-
-  if (!userInput.author) {
-    return res.send({ message: "something went wrong big time - author" });
-  }
-
-  if (!userInput.person) {
-    return res.send({ message: "something went wrong big time - person" });
-  }
-
-  async function addMemory(model, input): Promise<any> {
-
-    const newMemory = await model.create({
-      title: input.title,
-      text: input.text,
-      author: input.author,
-      person: input.person,
-    });
-    return newMemory;
-  }
-
-  async function findPersonAndUpdateMemories(model, id): Promise<any> {
-    const newMemory = await addMemory(Memory, userInput);
-    const updatedPerson = await model.findByIdAndUpdate(id, {
-       $push: { memories: newMemory },
-    });
-    return updatedPerson;
-  }
-
-  const updatedPerson = await findPersonAndUpdateMemories(Person, userInput.person);
-
-  res.send({ update: updatedPerson });
+    const id = req.params;
+    const userInput = req.body;
+
+    if (!id) {
+        return res.send({ message: "something went wrong big time" });
+    }
+
+    if (!userInput.title) {
```

Comment:
Look into `yup`: https://dev.to/franciscomendes10866/schema-validation-with-yup-and-express-js-3l19